src/adr/0040-generalize-event-bus.md

```diff
@@ -0,0 +1,27 @@
+#40. Generalize EventBus
+
+Date: 2020-06-11
+
+## Status
+
+Proposed
+
+## Context
+
+When searching for users by quota ratio if we set the value of the parameters to 0, the search feature is supposed to return newly created users
+who have not received any email yet at that point. However, this is not the case due to the quotas are currently being initialized only after
+an user receive the first email.
+
+In order to solve this problem, we need to initialize user quotas upon creation time. Event driven system allows indirection so that the users api 
```

Comment:
> Event driven system allows indirection so that the users api don't need a dependency on the quota components

Is it still part of the "Context" section? It's a general information about Event-Driven system contextualized to the case of user management and quota-search feature: it presents a solution to a problem that is not stated yet.

> The EventBus components currently defined in in mailbox-api would be suitable for this purpose.

Same here: we talk about a solution in the context section and yet we don't know what the problem is.

Let me try to write it another way (at this point we stated the problem is about initializing the searchable-per-user-quota):

The root of this problem is: there's currently no event at user creation and as the quota-search feature is a plugin of James, it can't be hardwire into the domain logic of user management to initialize the quota for a just created user.